homeassistant/components/tts/__init__.py

```diff
@@ -140,12 +140,12 @@ def __init__(
         self._data_gen = data_gen
         self._partial_data: list[bytes] | None = None
         self._error: Exception | None = None
-        self._listeners: list[asyncio.Queue[bytes]] | None
+        self._listeners: list[asyncio.Queue[bytes | None]] | None
 
     async def async_load_data(self) -> bytes:
         """Load the data from the generator."""
         if self._result_data is not None or self._partial_data is not None:
-            raise ValueError("Data already being loaded")
+            raise RuntimeError("Data already being loaded")
```

Comment:
I think we need to describe `self._result_data`, `self._partial_data` and the data in the queues in `self._listeners`, and their relationship and use at different times more, in a docstring, perhaps in the class. These three attributes handle the same data, but timing seems important, and that should be clarified, I think.

Reply:
Done. Moved them to class level and added doc strings.